src/unix/signal.c

```diff
@@ -63,6 +74,22 @@ static void uv__signal_global_init(void) {
 }
 
 
+static void uv__signal_global_reinit(void) {
+  /* We can only use signal-safe functions here.
+   * That includes read/write and close, fortunately.
+   * We do all of this directly here instead of resetting
+   * uv__signal_global_init_guard because
+   * uv__signal_global_once_init is only called from uv_loop_init
+   * and this needs to function in existing loops.
+   */
+  uv__close(uv__signal_lock_pipefd[0]);
+  uv__signal_lock_pipefd[0] = -1;
+  uv__close(uv__signal_lock_pipefd[1]);
+  uv__signal_lock_pipefd[1] = -1;
+  uv__signal_global_init();
+}
```

Comment:
I still need to be able to distinguish the initial case to know when to call `pthread_atfork`. That's currently being done by using the initial 0 value. If we initialize to -1, I lose that ability (can't distinguish initial from closed states). Now, we could make `uv__signal_global_reinit` *not* call `uv__signal_global_init` and just duplicate the code inline, but that's a pretty bad code smell. Or we could introduce another static variable, but that also seems like overkill...plus, if we're concerned that the `uv__signal_global_init_guard` `pthread_once_t` value will somehow get reset back to `PTHREAD_ONCE_INIT` in the child process (that's the only way that `pthread_once` would be able to call the function a second time), then we should also be concerned about the contents of any of our other static variables so we couldn't trust it.